lib/internal/Magento/Framework/Cache/Backend/RemoteSynchronizedCache.php

```diff
@@ -231,20 +231,9 @@ public function test($id)
      */
     public function save($data, $id, $tags = [], $specificLifetime = false)
     {
-        $dataToSave = $data;
-        $remHash = $this->loadRemoteDataVersion($id);
-        $isRemoteUpToDate = false;
-        if ($remHash !== false && $this->getDataVersion($data) === $remHash) {
-            $remoteData = $this->remote->load($id);
-            if ($remoteData !== false && $this->getDataVersion($data) === $this->getDataVersion($remoteData)) {
-                $isRemoteUpToDate = true;
-                $dataToSave = $remoteData;
-            }
-        }
-        if (!$isRemoteUpToDate) {
-            $this->remote->save($data, $id, $tags, $specificLifetime);
-            $this->saveRemoteDataVersion($data, $id, $tags, $specificLifetime);
-        }
```

Comment:
Hello @georgebabarus,

Can you please let us know the reason behind removing this code?

Reply:
@engcom-Hotel 

The code is not useful: if the version is correct and the data is version is correct -> the save method is not called, because this is **a cache hit by definition.**

The purpose of the cache is to avoid generating the data, all the consistency check should be done in the load method which is preventing the data to be generated. 

Once the data is generated I think is better to save it since the heavy processing was already done before the save action:
1. consistency check with Remote storage, done in the load method -> this should say what a cache hit means
2. data generated